src/hip_module.cpp

```diff
@@ -376,6 +376,7 @@ hipError_t ihipExtLaunchMultiKernelMultiDevice(hipLaunchParams* launchParamsList
 
         if(globalWorkSizeX > UINT32_MAX || globalWorkSizeY > UINT32_MAX || globalWorkSizeZ > UINT32_MAX)
         {
+            free(kds);
```

Comment:
it would be better to change `kds` into a `std::vector` then we don't need to explicitly free it

Reply:
Thanks, updated the PR accordingly